### DISCUSSION TITLE
Offer interfaces to other charts or map libraries

### DISCUSSION THREAD

**Discussion by ycli0536** [perspective-dev/perspective#2655] [Q&A]
### Description of Problem:

I really like the freedom you give in analyzing and presenting the data! I hope to have more features in visualizations, which are also already available in many existing libraries. And we don't have to recreate wheels.

I probably don't need most plugins for my data. There are a few examples as follow:
- show error bar in my scatter chart
- more advanced cursor operations, e.g., pan and zoom (x-only, y-only, omni, box) in charts
- change the basemap when I use map scatter

### Potential Solutions:

Don't know if there is any easy way for improvements. But can you provide interfaces to other data visualization tools or a blank component/widget that can accept and display other charts or map libraries so that I can display our data with them? And the user themselves should figure out how to parse the data between the data format used by perspective and the format required by other libraries.

**Comment by texodus** [comment]
> * show error bar in my scatter chart
> * more advanced cursor operations, e.g., pan and zoom (x-only, y-only, omni, box) in charts

If you have the time to add these, please consider [contributing them to the project](https://github.com/finos/perspective/blob/master/CONTRIBUTING.md) so the community can benefit. Perspective has an _unstable_ [API for plugins](https://github.com/finos/perspective/blob/master/rust/perspective-viewer/src/ts/plugin.ts) that the included visualizations already, but it is _unstable_ (e.g. we plan to change it) and complicated (it is not just a matter of drawing the data to a canvas), building a plugin from scratch is overkill.

> * change the basemap when I use map scatter

[Use `--map-tile-url` CSS variable](https://github.com/finos/perspective/blob/master/packages/perspective-viewer-openlayers/src/themes/material.dark.less#L32)

**Reply by ycli0536** [reply]
Yeah I don't think building a plugin from scratch (or just adding these advanced functions) is worthwhile too. Because there are many well-established libraries capable handle them very well. That's why I think a blank web widget might be a solution (also could be difficult). Thanks for bringing this API here and I'll try it! But sorry I don't know much about js.

Right now combining some different libraries together probably a workaround for me.
